pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/default.nix

```diff
@@ -88,16 +72,23 @@ stdenv.mkDerivation (finalAttrs: {
       --replace-fail '"libasound.so.2"' '"${alsa-lib}/lib/libasound.so.2"'
     substituteInPlace Telegram/ThirdParty/libtgvoip/os/linux/AudioPulse.cpp \
       --replace-fail '"libpulse.so.0"' '"${libpulseaudio}/lib/libpulse.so.0"'
+  '' + lib.optionalString (stdenv.hostPlatform.isLinux && withWebKitGTK) ''
     substituteInPlace Telegram/lib_webview/webview/platform/linux/webview_linux_webkitgtk_library.cpp \
-      --replace-fail '"libwebkitgtk-6.0.so.4"' '"${webkitgtk_6_0}/lib/libwebkitgtk-6.0.so.4"'
+      --replace-fail '"libwebkit2gtk-4.1.so.0"' '"${webkitgtk_4_1}/lib/libwebkit2gtk-4.1.so.0"'
   '' + lib.optionalString stdenv.hostPlatform.isDarwin ''
     substituteInPlace Telegram/lib_webrtc/webrtc/platform/mac/webrtc_environment_mac.mm \
       --replace-fail kAudioObjectPropertyElementMain kAudioObjectPropertyElementMaster
   '';
 
-  # We want to run wrapProgram manually (with additional parameters)
+  # Avoid double-wrapping
   dontWrapGApps = true;
-  dontWrapQtApps = true;
+
+  # Wrapping the inside of the app bundles, avoiding double-wrapping
+  dontWrapQtApps = stdenv.hostPlatform.isDarwin;
+
+  qtWrapperArgs = lib.optionals (stdenv.hostPlatform.isLinux && withWebKitGTK) [
+    ''"''${gappsWrapperArgs[@]}"''
+  ];
```

Comment:
For reasons this is not working.

Reply:
Are you sure? Works fine for me

Reply:
Give me a second, I might be looking at the wrong binary.

Reply:
Indeed, it's not present in the wrapper, weird it seemed to successfully loading the page for me, just not rendering it due to `Failed to open GBM device` over SSH X forwarding...

Reply:
Now it's set